r1cs/gadgets/std/src/fields/nonnative/nonnative_field_mul_result_gadget.rs

```diff
@@ -162,6 +213,13 @@ impl<SimulationF: PrimeField, ConstraintF: PrimeField>
         &self,
         mut cs: CS,
     ) -> Result<NonNativeFieldGadget<SimulationF, ConstraintF>, SynthesisError> {
+        let params = get_params(SimulationF::size_in_bits(), ConstraintF::size_in_bits(), SimulationF::Params::DIFFERENCE_WITH_HIGHER_POWER_OF_TWO);
+        if super::is_pseudo_mersenne::<SimulationF>() {
+            let mut result = self.to_non_native_field_gadget(&params);
```

Comment:
Even if with the current implementation it is clear that we never experience a mul result having a non-vanishing upper half of the limbs: wouldn't it be good practice to put an assert before or insert `to_non_native_field_gadget()`?

Reply:
Agree. Following the suggestion of Daniele I removed the `to_non_native_field_gagdet` function and placed the code in the `reduce` function of `NonNativeFieldMulResultGadget`, which is the only place where it is currently used, but I also added the assertion you suggested before doing the conversion.